### DISCUSSION TITLE
Add display of kbd tag in "View Mode"

### DISCUSSION THREAD

**Discussion by irmod** [damionx7/Zettel-Notes-Documentation#172] [Feature Requests]
I understand the limitations of the Markwon library, so I ask you to consider adding "View Mode" displays to the `<kbd>` tag.  I have a lot of notes that contain hotkeys or key combinations.

 I suggest using existing variables to display the `<kbd>` tag:
 - for the text font - "Code Font";
 - for text color - "Highlight Background";
 - for the background color - "Highlight Foreground";
 - rounding radius - 4px;
 - for example, `padding: 0.1em 0.25em;`

Using variables will allow you to invert the colors when the Theme changes _Night_ --> _Day_ --> _Night_

![Add_style_kbd_01.jpg](https://github.com/user-attachments/assets/9dd13e96-42d3-4914-a11a-2dfbfd6d1173)

![Add_style_kbd_02.jpg](https://github.com/user-attachments/assets/b3758464-9588-4ddb-802e-b2b001445b6d)

**Comment by damionx7** [comment]
Hey! Can you share the note file with `<kbd>` tag.

**Reply by irmod** [reply]
---
title: "The same Keyboard Shortcuts by default"
date: 2024-10-19
tags: zettel_notes, issue
---

# The same Keyboard Shortcuts by default

Source:  
<https://github.com/damionx7/Zettel-Notes-Documentation/issues/171>

*Settings* --> *Keyboard Shortcuts*

**Insert Tags**  

<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>T</kbd>

**Templates**  

<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>T</kbd>

You probably need to change one of the default combinations.

```
---
title: "The same Keyboard Shortcuts by default"
date: 2024-10-19
tags: zettel_notes, issue
---

# The same Keyboard Shortcuts by default

Source:  
<https://github.com/damionx7/Zettel-Notes-Documentation/issues/171>

*Settings* --> *Keyboard Shortcuts*

**Insert Tags**  

<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>T</kbd>

**Templates**  

<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>T</kbd>

You probably need to change one of the default combinations.
```

**Comment by irmod** [comment]
Here is an example of a note that contains `<kbd>` tags

**Comment by damionx7** [comment]
I have added this feature. Check out the screenshot.

![Screenshot_2024-10-27-17-20-14-84_377fe8767025e4fa6dddb62d6df8f716](https://github.com/user-attachments/assets/d4ce7630-cbc7-4206-affd-6cf6954d40ee)

**Comment by irmod** [comment]
@damionx7
Your screenshot shows exactly what I assume is in the new version of Zettel Notes

I'm waiting for the new version

**Comment by irmod** [comment]
@damionx7, I tested on version 2.7.5.  Here are my results on

- Zettel Notes v.2.7.5
- Xiaomi Redmi Note 7 (API 28)
- 64-bit ARMv8-A
- Android 9 (Pie)

![Insert_kbd_Zettel_Notes_02.png](https://github.com/user-attachments/assets/2ccca12c-b555-44a4-98c7-b15eddd76774)

Edit Mode:

![Insert_kbd_Zettel_Notes_Edit_02.png](https://github.com/user-attachments/assets/d8fd936c-7916-4a91-bcba-eb932cf486d2)

Light:

![Insert_kbd_Zettel_Notes_01.png](https://github.com/user-attachments/assets/028d1fb4-ce30-4a66-afe0-af8c34158aff)

For example on other applications:

![Insert_kbd_Markor_03.png](https://github.com/user-attachments/assets/3e3d69dd-3e13-4a8d-96e4-4fe2f0ba8739)

![Insert_kbd_Obsidian_06.png](https://github.com/user-attachments/assets/2536b3db-c17e-47c5-860f-bc2c12d91b5d)

![Insert_kbd_Epsilon_05.png](https://github.com/user-attachments/assets/63002f17-4537-47d1-9df2-4a24d76f23b5)

Looks like I have something wrong...

**Reply by damionx7** [reply]
Can you share this note file.

**Reply by irmod** [reply]
@damionx7 

I changed the theme to one of the preset ones.  Tag highlighting works.

What variables did you use for the foreground and background lights?

![Screenshot_2024-10-30-21-52-25-734_org.eu.thedoc.zettelnotes.png](https://github.com/user-attachments/assets/81ecccbf-df13-487c-b632-42b3eb017f8b)

![Screenshot_2024-10-30-21-56-58-514_org.eu.thedoc.zettelnotes.png](https://github.com/user-attachments/assets/97925388-dbb4-4217-abd5-dc9350421f47)

**Comment by irmod** [comment]
Looks like I'm confused...

@damionx7, Please specify how you set the colors:

Theme **Day**

`==markText==`
Foreground => "Highlight Foreground"
Background => "Highlight Background"

`<mark>markText</mark>`
Foreground => "Highlight Foreground"
Background => "Highlight Background"

`<kbd>key</kbd>`
Foreground => "_Background Color_" ???
Background => "_Text Color_" ???

Theme **Night**

`==markText==`
Foreground => "Highlight Foreground"
Background => "Highlight Background"

`<mark>markText</mark>`
Foreground => "Highlight Foreground"
Background => "Highlight Background"

`<kbd>key</kbd>`
Foreground => "_Background Color_" ???
Background => "_Text Color_" ???

**Comment by irmod** [comment]
@damionx7 

I showed screenshots of the Night Theme for the installed Markdown Style "Hertz Dark"

**Reply by damionx7** [reply]
Hey, I was using the same settings that you suggested in your first post. 
```
for text color - "Highlight Background";
for the background color - "Highlight Foreground";
```

**Comment by irmod** [comment]
@damionx7, I tested the `kbd` tag highlighting on 14 themes (13 built-in and Hertz Dark) in Day and Night mode.  
On themes in Night mode, the background of the `kbd` tag matches the color of the main background of the note.

![Day_v2.7.5.jpg](https://github.com/user-attachments/assets/2ef4daf7-15cf-4f4f-8b0b-b90e948e4ce6)

![Night_v2.7.5.jpg](https://github.com/user-attachments/assets/8929e1e7-6186-469b-baae-ff43d1340ee1)


To correctly display tag highlighting when switching the theme between Day <--> Night in applications (such as Obsidian) and frameworks (Bootstrap, UiKit) set the tag text color through the main background variable, and the tag background color through the main text variable.

```css
kbd {
  font-family: var(--bs-font-monospace);
  font-size: 1em;
}

kbd {
  padding: 0.1875rem 0.375rem;
  font-size: 0.875em;
  color: var(--bs-body-bg);
  background-color: var(--bs-body-color);
  border-radius: 0.25rem;
}
```

You might agree that in the case of Zettel Notes it would be correct to set the text color of the `kbd` tag through the note's body text variable, which can be overridden by the "Text Color" from Markdown Style.  The background color of the `kbd` tag is set via a variable from the theme design Theme --> Customize "Background Color".

It would be great if you could add to the Markdown Style settings the ability to redefine the background of a note to an arbitrary color, as is done for other elements.

Then I can add the original "Catppuccin" family themes to Zettel Notes 🙂

![Catppuccin.png](https://github.com/user-attachments/assets/a82fc50b-7ebb-455e-b65f-72050a414ad4)

**Reply by damionx7** [reply]
I am amazed by your extensive testing :+1:. 

Zettel notes currently uses [markwon](https://github.com/noties/Markwon/issues) library for viewer which is based on text view and doesn't support css.

I will add background color option in markdown style. For now, I will match kbd tag with highlight colors.

**Comment by irmod** [comment]
@damionx7, I would be very grateful if you could implement color selection for Background in Markdown Style 🙏
